topics/statistics/tutorials/RNN/tutorial.md

```diff
@@ -0,0 +1,365 @@
+thousands of layers. 
+
+# Text representation schemes
+
+In this tutorial we perform sentiment analysis on IMDB (https://www.imdb.com/) movie reviews dataset ({% cite maas-EtAl %}). We train our RNN on 
+the training dataset, which is made up of 25000 movie reviews, some positive and some negative. We then test our RNN on the test set, which is 
+also made up of 25000 movie reviews, again some positive and some negative. The training and test sets have no overlap. Since we are dealing with 
+text data, it's a good idea to review various mechanisms for representing text data. Before that, we are going to briefly discuss how to preprocess 
+text documents. 
+
+## Text preprocessing
+
+The first step is to tokenize a document, i.e., break it down into words. Next, we remove the punctuation marks, URLs, and stop words -- words like 
+'a', 'of', 'the', etc. that happen frequently in all documents and do not have much value in discriminating between documents. Next, we normalize 
+the text, e.g., replace 'brb' with 'Be right back', etc. Then, We then run the spell checker to fix typos and also make all words lowercase. Next, we perform stemming or lemmatization. Basically, if we have words like 'organizer', 'organize', 'organized', and 'organization' we want to reduce all of them to a single word. Stemming cuts the end of these words to come up with a single root (e.g., 'organiz'). The root may not be an actual word. 
+Lemmatization is smarter in that it reduces the word variants to a root that is actually a word (e.g., 'organize'). All of these steps help reduce 
+the number of features in feature vector of a document and should make the training of our model faster/easier.
+
+For this introductory tutorial, we do minimal text preprocessing. We ignore the top 50 words in IMDB reviews (mostly stop words) and include 
+the next 10,000 words in our dataset. Reviews are limited to 500 words. They are trimmed if they are longer and padded if they are shorter.
+ 
+## Bag of words and TF-IDF
+
+If you don't care about the order of the words in a document, you can use bag of words (BoW) or text frequency inverse document frequency (TF-IDF).
+In these models we have a 2 dimensional array. The rows represent the documents (in our example, the movie reviews) and the columns
+represent the words in our vocabulary (all the unique words in all the documents). If a word is not present in a document, we have a zero 
+at the corresponding row and column as the entry. If a word is present in the document, we have a one as the entry -- Alternatively, we could use 
+the word count or frequency.
+
+![Alternative text](../../images/BoW.png "Bag of words (BoW) representation")
+
+Suppose we have the following 2 documents: 1) Magic passed the basketball to Kareem, and 2) Lebron stole the basketball from Curry. The BoW 
+representation of these documents is given in Figure 10. 
```

Comment:
I'm slightly worried about the figure references (not so many others do this) if you want to e.g. add a picture at the top, but we can fix that when it becomes an issue maybe.

Reply:
Ok, so I leave it as is for now.